profile/profile.js

```diff
@@ -0,0 +1,361 @@
+					$("#lastNameText").html($("#lastNameInput").val());
+					$("#emailText").html($("#emailInput").val());
+					$("#phoneNumberText").html($("#phoneNumberInput").val());
+
+					$(".personal-info").find('p').show();		
+					$(".personal-info").find('input').hide();
+
+					$("#personalInformationSaveButton").hide();
+					$("#personalInformationCancelButton").hide();
+					$("#personalInformationEditButton").show();
+				} else {
+					alert(response.error);
+				}
+
+				$("#personalInformationSaveButton").prop('disabled', false);
+				$("#personalInformationCancelButton").prop('disabled', false);
+			},
+			error: function(response) {
+				alert("Unable to communicate with the server. Please try again later.");
+				$("#personalInformationSaveButton").prop('disabled', false);
+				$("#personalInformationCancelButton").prop('disabled', false);
+			}
+		});
+	});
+
+	$("#addShiftButton").click(function(e) {
+		let shiftRow = $('<div></div>').addClass("add-shift-div");
+		let siteSelect = $('<select></select>').addClass("siteSelect");
+		let dateSelect = $('<select></select>').addClass("dateSelect").attr('disabled', true);
+		let timeSelect = $('<select></select>').addClass("timeSelect").attr('disabled', true);
+		
+		siteSelect.append($('<option disabled selected value="" style="display:none"> -- Select a site -- </option>'));
+		dateSelect.append($('<option disabled selected value="" style="display:none"> -- Select a date -- </option>'));
+		timeSelect.append($('<option disabled selected value="" style="display:none"> -- Select a time -- </option>'));
+
+		for (const [siteId, title] of sitesMap) {
+			siteSelect.append($('<option>', {
+				value: siteId,
+				text: title
+			}));
+		}
+
+		siteSelect.change(function() {
+			timeSelect.children('option').remove();
+			timeSelect.append($('<option disabled selected value="" style="display:none"> -- Select a time -- </option>'));
+			timeSelect.attr('disabled', true);			
+			dateSelect.children('option').remove();
+			dateSelect.append($('<option disabled selected value="" style="display:none"> -- Select a date -- </option>'));
+			dateSelect.attr('disabled', false);
+			
+			let siteId = $(this).val();
+
+			// populate dates select now based on which site was chosen
+			for (const dateString of shiftsMap.get(siteId).keys()) {
+				dateSelect.append($('<option>', {
+					value: dateString,
+					text: dateString
+				}));
+			}
+		});
+
+		dateSelect.change(function() {
+			timeSelect.children('option').remove();
+			timeSelect.append($('<option disabled selected value="" style="display:none"> -- Select a time -- </option>'));			
+			timeSelect.attr('disabled', false);
+			let siteId = siteSelect.val();
+			let dateString = $(this).val();
+
+			// populate shift times select now based on which date was chosen AND the user is not already signed up for it
+			for (const shift of shiftsMap.get(siteId).get(dateString)) {
+				if (shift.signedUp) continue;
```

Comment:
Rather than simply not showing the shift times that the user has signed up for, I would suggest disabling them. You can just add ```disabled: shift.signedUp ? true : false``` It will show the shift time but have it be disabled. I think this is more intuitive than simply having an empty drop-down.